src/gnu/prolog/io/TermWriter.java

```diff
@@ -473,7 +473,14 @@ protected void displayAtom(WriteOptions options, AtomTerm atom)
 		if (options.quoted)
 		{
 			String value = atom.value;
-			print(needBeQuoted(value) ? getSingleQuoted(value) : value);
```

Comment:
I think you want to put this test in the needBeQuoted function rather than here.